app/views/ReportView/components/PathwayAnalysis/components/Pathway/index.tsx

```diff
@@ -62,20 +64,28 @@ const Pathway = ({
       newPathway.set('legend', 'v2');
 
       let pathwayResp: PathwayImageType;
+      let pathwayCall;
+
       if (initialPathway) {
-        pathwayResp = await api.put(
+        pathwayCall = api.put(
           `/reports/${report.ident}/summary/pathway-analysis`,
           newPathway,
           {},
           true,
-        ).request(isSigned);
+        );
       } else {
-        pathwayResp = await api.post(
+        pathwayCall = api.post(
           `/reports/${report.ident}/summary/pathway-analysis`,
           newPathway,
           {},
           true,
-        ).request(isSigned);
+        );
+      }
+
+      if (isSigned) {
+        showConfirmDialog(pathwayCall);
+      } else {
+        pathwayResp = await pathwayCall.request();
       }
 
       setPathwayImage(pathwayResp);
```

Comment:
missing `showConfirmDialog` in dependency array, and move line 91 to 94 inside the else statement